crates/bevy_core_pipeline/src/prepass/mod.rs

```diff
@@ -0,0 +1,147 @@
+//! Run a prepass before the main pass to get the depth and/or normals texture.
+//! The depth prepass texture is then used by the main pass to reduce overdraw.
+//!
+//! To enable the prepass, you need to add a `PrepassSettings` component to a `Camera`.
+//! Both textures are available on the `PrepassTextures` component attached to each `Camera` with a `PrepassSettings`
+//!
+//! Currently only works for 3d
+
+pub mod node;
+
+use bevy_ecs::prelude::*;
+use bevy_render::{
+    render_phase::{CachedRenderPipelinePhaseItem, DrawFunctionId, EntityPhaseItem, PhaseItem},
+    render_resource::{CachedRenderPipelineId, Extent3d},
+    texture::CachedTexture,
+};
+use bevy_utils::FloatOrd;
+
+/// Add a [`PrepassSettings`] component to a [`crate::prelude::Camera3d`] to perform a depth and/or normal prepass.
+/// These textures are useful for various screen-space effects and reducing overdraw in the main pass.
+///
+/// The textures are automatically added to the default mesh view bindings. You can also get the raw textures
+/// by querying the [`ViewPrepassTextures`] component on the camera with the [`PrepassSettings`].
+///
+/// When using the default mesh view bindings you should be able to use `prepass_depth()`
+/// and `prepass_normal()` to load the related textures. These functions are defined in `bevy_pbr::utils`.
+/// See the `shader_prepass` example that shows how to use it.
+#[derive(Clone, Component)]
+pub struct PrepassSettings {
+    /// If true then depth values will be copied to a separate texture available to the main pass.
+    /// This will help reduce overdraw in the main pass.
```

Comment:
the base depth buffer should be used in the main pass regardless of whether we make a separate copy available to read from as a texture.

perhaps the field should be renamed to "depth_texture_enabled"

Reply:
i thought i was adding suggestions as well but apparently not ... i hope the others are all clear but here i would suggest something like "overdraw in the main pass will always be reduced, whether a readable depth texture copy is generated or not."

Reply:
The way I see it, there's some amount of overdraw being reduced because we use a depth texture without a prepass, but an empty prepass won't reduce more overdraw than just disabling the prepass right? My understanding is that having a depth prepass can help remove more overdraw though.

Reply:
I changed it to "The main pass already uses a depth texture by default which helps reduce overdraw, but this will help reduce it even more." Is this better?